src/engine/plugins/nearest.cpp

```diff
@@ -44,14 +44,22 @@ Status NearestPlugin::HandleRequest(const api::NearestParameters &params,
     // We want to keep the user-interface as is, so make a copy. Should be cheap.
     auto constraint_params = params;
 
-    const auto max_radius_m = static_cast<double>(max_radius);
+    if (max_radius != -1)
+    {
+        const auto max_radius_m = static_cast<double>(max_radius);
+
+        if (constraint_params.radiuses.empty())
+            constraint_params.radiuses.push_back(boost::make_optional(max_radius_m));
+        else if (constraint_params.radiuses[0])
+            boost::algorithm::clamp(*constraint_params.radiuses[0], 0., max_radius_m);
+    }
 
-    if (constraint_params.radiuses.empty())
-        constraint_params.radiuses.push_back(boost::make_optional(max_radius_m));
-    else if (constraint_params.radiuses[0])
-        boost::algorithm::clamp(*constraint_params.radiuses[0], 0., max_radius_m);
+    if (max_results != -1)
+    {
+        constraint_params.number_of_results = boost::algorithm::clamp(params.number_of_results, 0, max_results);
```

Comment:
All other plugins return a `TooBig` status code in case the user-provided params are over the global system's limit (e.g. [viaroute](https://github.com/Project-OSRM/osrm-backend/blob/938dff011fb52fedb8b787d7403b0c8fb1d67f5a/src/engine/plugins/viaroute.cpp#L35-L43)). I think in the case of `max_results` we should do the same, since this specifically is `TooBig`'s raison d’être.